anndata/_core/aligned_mapping.py

```diff
@@ -249,7 +249,7 @@ def __init__(
 
     @property
     def dim_names(self) -> pd.Index:
-        return self.parent_mapping.dim_names[self.subset_idx]
+        return super().dim_names[self.subset_idx]
```

Comment:
I don't think you need to redefine this property, since the indices of the `parent` should already be subsetted (the parent `AnnData` of a `AlignedView` should always be a view)